File: _work/backend-watchdog-full-repair-02/src/iii-sandbox/packages/sdk-rust/src/network.rs
use std::sync::Arc;

use crate::client::HttpClient;
use crate::error::Result;
use crate::types::{
    NetworkConnectResponse, NetworkDeleteResponse, NetworkDisconnectResponse, NetworkListResponse,
    SandboxNetwork,
};

pub struct NetworkManager {
    client: Arc<HttpClient>,
}

impl NetworkManager {
    pub fn new(client: Arc<HttpClient>) -> Self {
        Self { client }
    }

    pub async fn create(
        &self,
        name: &str,
        driver: Option<&str>,
    ) -> Result<SandboxNetwork> {
        self.client
            .post(
                "/sandbox/networks",
                Some(&serde_json::json!({
                    "name": name,
                    "driver": driver,
                })),
            )
            .await
    }

    pub async fn list(&self) -> Result<NetworkListResponse> {
        self.client.get("/sandbox/networks").await
    }

    pub async fn connect(
        &self,
        network_id: &str,
        sandbox_id: &str,
    ) -> Result<NetworkConnectResponse> {
        self.client
            .post(
                &format!("/sandbox/networks/{network_id}/connect"),
                Some(&serde_json::json!({ "sandboxId": sandbox_id })),
            )
            .await
    }

    pub async fn disconnect(
        &self,
        network_id: &str,
        sandbox_id: &str,
    ) -> Result<NetworkDisconnectResponse> {
        self.client
            .post(
                &format!("/sandbox/networks/{network_id}/disconnect"),
                Some(&serde_json::json!({ "sandboxId": sandbox_id })),
            )
            .await
    }

    pub async fn delete(&self, network_id: &str) -> Result<NetworkDeleteResponse> {
        self.client
            .delete(&format!("/sandbox/networks/{network_id}"))
            .await
    }
}

File: Agente Yaiwes principal/_work/backend-watchdog-full-repair-02/src/iii-sandbox/packages/sdk-python/src/iii_sandbox/network.py
from __future__ import annotations

from .client import HttpClient
from .types import SandboxNetwork


class NetworkManager:
    def __init__(self, client: HttpClient):
        self._client = client

    async def create(
        self, name: str, driver: str | None = None
    ) -> SandboxNetwork:
        body: dict = {"name": name}
        if driver is not None:
            body["driver"] = driver
        data = await self._client.post("/sandbox/networks", body)
        return SandboxNetwork(**data)

    async def list(self) -> dict:
        return await self._client.get("/sandbox/networks")

    async def connect(
        self, network_id: str, sandbox_id: str
    ) -> dict:
        return await self._client.post(
            f"/sandbox/networks/{network_id}/connect",
            {"sandboxId": sandbox_id},
        )

    async def disconnect(
        self, network_id: str, sandbox_id: str
    ) -> dict:
        return await self._client.post(
            f"/sandbox/networks/{network_id}/disconnect",
            {"sandboxId": sandbox_id},
        )

    async def delete(self, network_id: str) -> dict:
        return await self._client.delete(
            f"/sandbox/networks/{network_id}"
        )
